doc/developer-guides/coding_guidelines.rst

```diff
@@ -1121,6 +1428,312 @@ Compliant example::
+       enum enum_showcase showcase;
+       
+       switch (showcase) {
+       case ENUM_SHOWCASE_0:
+           /* showcase */
+           break;
+       case 1U:
+           /* showcase */
+           break;
+       default:
+           /* showcase */
+           break;
+
+
+TY-10: const qualifier shall not be discarded in cast operation
+===============================================================
+
+Compliant example::
+
+    const uint32_t *showcase_const;
+    const uint32_t *showcase = showcase_const;
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       const uint32_t *showcase_const;
+       uint32_t *showcase = (uint32_t *)showcase_const;
+
+
+TY-11: A variable shall be declared as static if it is only used in one file
+============================================================================
+
+Compliant example::
+
+    /* In `showcase.c`: */
+    /* `showcase` is only in `showcase.c` */
+    static uint32_t showcase;
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       /* In `showcase.c`: */
+       /* `showcase` is only in `showcase.c` */
+       uint32_t showcase;
+
+
+TY-12: The type conversion shall be explicit
+============================================
+
+Compliant example::
+
+    uint32_t showcase_u32;
+    uint64_t showcase_u64 = 64UL;
+    
+    showcase_u32 = (uint32_t)showcase_u64;
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       uint32_t showcase_u32;
+       uint64_t showcase_u64 = 64UL;
+       
+       showcase_u32 = showcase_u64;
+
+
+TY-13: A complex integer expression shall not be cast to a wider type
+=====================================================================
+
+Compliant example::
+
+    uint32_t showcase_u32_1 = 10U;
+    uint32_t showcase_u32_2 = 10U;
+    uint64_t showcase_u64;
+    
+    showcase_u64 = (uint64_t)showcase_u32_1 + (uint64_t)showcase_u32_2;
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       uint32_t showcase_u32_1 = 10U;
+       uint32_t showcase_u32_2 = 10U;
+       uint64_t showcase_u64;
+       
+       showcase_u64 = (uint64_t)(showcase_u32_1 + showcase_u32_2);
+
+
+TY-14: A complex signed integer expression shall not be cast to unsigned type
+=============================================================================
+
+Compliant example::
+
+    int32_t showcase_signed_1 = 10;
+    int32_t showcase_signed_2 = 5;
+    uint32_t showcase_unsigned;
+    
+    showcase_unsigned = (uint32_t)showcase_signed_1 - (uint32_t)showcase_signed_2;
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       int32_t showcase_signed_1 = 10;
+       int32_t showcase_signed_2 = 5;
+       uint32_t showcase_unsigned;
+       
+       showcase_unsigned = (uint32_t)(showcase_signed_1 - showcase_signed_2);
+
+
+TY-15: A complex unsigned integer expression shall not be cast to signed type
+=============================================================================
+
```

Comment:
Similar comments for this rule:

Maybe add an explanation, something like:
```
Unsigned operands of an expression shall be cast to a signed type if the operation
results are needed as a signed type.
```

Reply:
Sorry for the confusion. Similar to the rule TY-13.
What I would like to express is:
Cast is allowed. But the cast shall be separate on each variable, rather than casting the full expression.